credentials/go10.go

```diff
@@ -0,0 +1,69 @@
+// +build go1.10
+
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package credentials
+
+import (
+	"fmt"
+	"net/url"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+// ParseSpiffeID parses the Spiffe ID from State and fill it into SpiffeID.
+// An error is returned only when we are sure Spiffe ID is used but the format
+// is wrong.
+// This function can only be used with go version 1.10 and onwards. When used
+// with a prior version, no error will be returned, but the field
+// TLSInfo.SpiffeID wouldn't be plumbed.
+func (t *TLSInfo) ParseSpiffeID() error {
```

Comment:
Do we expect users to call this?  This would now be part of our public API.  Can we make this a function in `internal/...` instead (`internal/credentials`?)?